File: Assignment24/C/Program5.c
// PS - Accept string from user and reverse the string in plcace

#include <stdio.h>

void strrevX(char *str)
{
    char *start = NULL;
    char *end = NULL;
    char temp = '\0';

    start = str;
    end = str;

    while(*end != '\0'){
        end++;
    }
    end--;

    while (start < end)
    {
        temp = *start;
        *start = *end;
        *end = temp;

        start++;
        end++;
    }
    
}

int main()
{
    char Arr[20];

    printf("Enter string \n");
    scanf("%[^'\n']s", Arr);

    strrevX(Arr);

    printf("Reverse sting is : %s\n", Arr);

    return 0;
}

File: Assignment33/Java/Program5.java
// PS - Write a java program which accept string from user and reverse the string

import java.util.*;

class StringX
{
    public String ReverseString(String str)
    {
        char Arr[] = str.toCharArray();

        int Start = 0;
        int End = Arr.length-1;

        char cTemp = ' ';

        while(Start < End)
        {
            cTemp = Arr[Start];
            Arr[Start] = Arr[End];
            Arr[End] = cTemp;

            Start++;
            End--;
        }

        return new String(Arr);
    }
}

class Program5
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter string");
        String content = sobj.nextLine();

        StringX obj = new StringX();
        String sRet = obj.ReverseString(content);

       System.out.println("Reverse string is :"+sRet);
    }
}